src/gluonts/env.py

```diff
@@ -11,13 +11,22 @@
 # express or implied. See the License for the specific language governing
 # permissions and limitations under the License.
 
+
 import os
+from .core.context import Context
+
+
+class Environment(Context):
+    # Maximum number of times a transformation can receive an input without
+    # returning an output. This parameter is intended to catch infinite loops
+    # or inefficiencies, when transformations never or rarely return
+    # something.
+    max_idle_transforms: int = os.environ.get(
+        "GLUONTS_MAX_IDLE_TRANSFORMS", "100"
+    )
+
+    # we want to be able to disable TQDM, for example when running in sagemaker
+    use_tqdm: bool = True
+
 
-"""
-Maximum number of times a transformation can receive an input without returning an output.
-This parameter is intended to catch infinite loops or inefficiencies, when transformations
-never or rarely return something.
-"""
-GLUONTS_MAX_IDLE_TRANSFORMS = int(
-    os.environ.get("GLUONTS_MAX_IDLE_TRANSFORMS", "100")
-)
+env = Environment()
```

Comment:
This could be `env = Context(max_idle_transforms=..., use_tqdm=...)`

The advantage of subclassing is the type annotation? What does that provide in practice?

Reply:
A more concise interface and type-checking. So now you get an error if you use a non-supported type.

Reply:
> So now you get an error if you use a non-supported type.

We could have a small test case for that? That can help understanding this use case, and also make sure pydantic is doing what it's supposed to be doing